Latest alpha version is less resilient to import errors

## 🐛 Bug Report

In version 44 a failed import reports an error but docusaurus continues to run and reload changes. In 48 the server terminates and has to be restarted after the error is fixed. This is quite inconvenient during the development/documentation cycle.

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## To Reproduce

Note that the example project below is using Lerna. This is just to make the repro easier -- our actual project doesn't use Lerna but exhibits the same behaviour.

1. Download / clone https://github.com/jugglingcats/docusaurus-repro
2. `lerna bootstrap`
3. `cd docusaurus-site`
4. `npm start`
5. Notice that the server starts, albeit with an error
```
Starting the development server...

× Client
  Compiled with some errors in 8.43s

i ｢wds｣: Project is running at http://localhost:3001/
i ｢wds｣: webpack output is served from /
i ｢wds｣: Content not from webpack is served from D:\alfie\tmp\docusaurus-repro\packages\docusaurus-site
i ｢wds｣: 404s will fallback to /index.html


./docs/mdx.md
Module not found: Can't resolve '@theme/invalid' in 'D:\alfie\tmp\docusaurus-repro\packages\docusaurus-site\docs'
```
6. Kill the server
7. Edit `package.json` to upgrade to version 48. Re-run `lerna bootstrap`
7. Try to start docusaurus

## Expected behavior

Docusaurus should start and report the import error.

## Actual Behavior

Docusaurus reports the error and terminates.
```
Starting the development server...

* Client █████████████████████████ after emitting (98%)


i ｢wds｣: Project is running at http://localhost:3000/
i ｢wds｣: webpack output is served from /
i ｢wds｣: Content not from webpack is served from D:\alfie\tmp\docusaurus-repro\packages\docusaurus-site
i ｢wds｣: 404s will fallback to /index.html
Client bundle compiled with errors therefore further build is impossible.
```

## Your Environment

- Docusaurus version used: 2.0.0-alpha.48
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0):
- Operating system and version (desktop or mobile):

## Reproducible Demo

https://github.com/jugglingcats/docusaurus-repro

See above for detailed steps.

--- a/packages/docusaurus/src/webpack/client.ts
+++ b/packages/docusaurus/src/webpack/client.ts
@@ -17,6 +17,7 @@ import LogPlugin from './plugins/LogPlugin';
 
 export function createClientConfig(props: Props): Configuration {
   const isProd = process.env.NODE_ENV === 'production';
+  const isBuilding = process.argv[2] === 'build';
   const config = createBaseConfig(props, false);
 
   const clientConfig = merge(config, {
@@ -32,26 +33,6 @@ export function createClientConfig(props: Props): Configuration {
       runtimeChunk: true,
     },
     plugins: [
-      // Plugin to force terminate building if errors happened in the client bundle
-      {
-        apply: compiler => {
-          compiler.hooks.done.tap('client:done', stats => {
-            if (stats.hasErrors()) {
-              console.log(
-                chalk.red(
-                  'Client bundle compiled with errors therefore further build is impossible.',
-                ),
-              );
-
-              stats.toJson('errors-only').errors.forEach(e => {
-                console.error(e);
-              });
-
-              process.exit(1);
-            }
-          });
-        },
-      },
       new ChunkAssetPlugin(),
       // Show compilation progress bar and build time.
       new LogPlugin({
@@ -59,6 +40,29 @@ export function createClientConfig(props: Props): Configuration {
       }),
     ],
   });
+
+  // When building include the plugin to force terminate building if errors happened in the client bundle.
+  if (isBuilding) {
+    clientConfig.plugins!.push({
+      apply: compiler => {
+        compiler.hooks.done.tap('client:done', stats => {
+          if (stats.hasErrors()) {
+            console.log(
+              chalk.red(
+                'Client bundle compiled with errors therefore further build is impossible.',
+              ),
+            );
+
+            stats.toJson('errors-only').errors.forEach(e => {
+              console.error(e);
+            });
+
+            process.exit(1);
+          }
+        });
+      },
+    });
+  }
 
   return clientConfig;
 }

